test/test_history.cpp

```diff
@@ -0,0 +1,31 @@
+#include <gtest/gtest.h>
+#include <fstream>
+
+#include "history.hpp"
+
+namespace mamba
+{
+    TEST(history, parse)
+    {
+        History original_history("original_history_data");
+        History aux_history("aux_history_data");
+
+        std::vector<History::UserRequest> user_reqs = original_history.get_user_requests();
+
+        std::ifstream original_history_file("original_history_data");
+        std::stringstream original_history_buffer;
+        original_history_buffer << original_history_file.rdbuf();
+
+        std::ifstream aux_history_file("aux_history_data");
+        std::stringstream aux_history_buffer;
+        aux_history_buffer << aux_history_file.rdbuf();
+
+        ASSERT_EQ(original_history_buffer.str(), aux_history_buffer.str());
```

Comment:
if those two buffers are empty, it will still compare true, right?
Can we add some checks to make sure we're actually comparing something here?

Reply:
I can do that, but isn't it redundant? `original_history_buffer` will never be empty, because there will always be content inside  the file `original_history_data`.

Reply:
have you tested what happens when you delete the file and it doesn't exist? it's possible that there is no loud error...

Reply:
have you tested what happens when you delete the file and it doesn't exist? it's possible that there is no loud error...

Reply:
Also the way I see it right now is that your aux_history_file is actually empty, so I am pretty sure you are comparing two empty strings here otherwise the test shouldn't pass :)

And you aren't using the `user_requests` that you read from the other history file as far as i can see.

Reply:
No I didn't, but the file is not going to be deleted by any "natural" cause, if that makes sense. Only if the user does something they shouldn't.
But I see your point, I'm going to test it and see what happens and depending on what happens I'll throw an error. Thanks for the feedback 🐺

Reply:
Ohh, yeah, you're totally right.